qiita_db/support_files/patches/python_patches/47.py

```diff
@@ -0,0 +1,30 @@
+from qiita_db.study import Study
+
+
+class ForRecursion(object):
+    """for some strange reason, my guess is how we are executing the patches
+    recursion doesn't work directly so decided to use a class to make it
+    work"""
+
+    @classmethod
+    def change_status(cls, artifact, status):
+        for a in artifact.children:
+            try:
+                a.visibility = status
+            except:
+                # print so we know which changes failed and we can deal by hand
+                print "failed aid: %d, status %s" % (artifact.id, status)
```

Comment:
Failed "aid?" Not sure I would understand what that was saying if I got this message.